src/components/ArrayInput/index.ts

```diff
@@ -117,17 +117,17 @@ class ArrayInput extends Element implements IFocusable, IBindable {
 
     protected _renderChanges: boolean;
 
-    constructor(args: ArrayInputArgs = {}) {
-        // remove binding because we want to set it later
-        const binding = args.binding;
-        delete args.binding;
-
+    constructor(args: Readonly<ArrayInputArgs> = {}) {
         const container = new Container({
             dom: args.dom,
             flex: true
         });
 
-        super(container.dom, args);
+        const elementArgs = { ...args, dom: container.dom };
```

Comment:
Think you want the order of `...args` and `dom:` swapped.

Reply:
Unless I'm misunderstanding things!

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals

Reply:
The code before forced the dom arg to the `container.dom` value. So we want to override the args object with the `container.dom` too. If the values were reversed, the args could potentially replace `container.dom`.

Reply:
Oic that's expected! 👍